webapp/components/product-info.html

```diff
@@ -64,44 +113,11 @@
       }
 
       parseProductSpec(spec) {
-        // @sha (optional)
-        let revision = '';
-        const atIndex = spec.indexOf('@');
-        if (atIndex > 0) {
-          revision = spec.substr(atIndex + 1);
-          spec = spec.substr(0, atIndex);
-        }
-        // [foo,bar] labels syntax (optional)
-        let labels = new Set();
-        const arrayIndex = spec.indexOf('[');
-        if (arrayIndex > 0) {
-          let labelsStr = spec.substr(arrayIndex + 1);
-          if (labelsStr[labelsStr.length - 1] !== ']') {
-            throw 'Expected closing bracket';
-          }
-          labelsStr = labelsStr.substr(0, labelsStr.length - 1);
-          labelsStr.split(',').forEach(l => labels.add(l));
-          spec = spec.substr(0, arrayIndex);
-        }
-        // product
-        const product = this.parseProduct(spec);
-        product.revision = revision;
-        product.labels = Array.from(labels);
-        return product;
+        return window.parseProductSpec(spec);
```

Comment:
I don't know much about the isolation of web components. Would these two methods leak into the global `window`?

Besides, why do we need to put them in the global namespace in the first place?

Reply:
I use them above to populate `DEFAULT_PRODUCTS`, and couldn't think how to refer to the function otherwise.